.github/scripts/upload_tests_results.py

```diff
@@ -0,0 +1,342 @@
+    tests_data_with_owners = []
+    for test in tests_data:
+        owners = []
+        target_path = test["suite_name"]
+        for path, path_owners in entries:
+            path = path + "/" if path[-1] != "/" else path
+            path = path + "*" if path[-1] != "*" else path
+            if fnmatch.fnmatch(f"/{target_path}", path):
+                owners.extend(path_owners)
+
+        test["owners"] = ",".join(owners)
+        tests_data_with_owners.append(test)
+    return tests_data_with_owners
+
+
+def upload_codeowners(session, entries):
+    # Вставка данных в таблицу
+    for entry in entries:
+        path, owners = entry
+        sql = f"""
+        UPSERT INTO codeowners (path, owners) VALUES 
+            ("{path}", {str(owners).replace("'", '"')});
+        """
+        session.transaction().execute(sql, commit_tx=True)
+
+
+def create_table(session, sql):
+
+    try:
+        session.execute_scheme(sql)
+    except ydb.issues.AlreadyExists:
+        pass
+    except Exception as e:
+        print(f"Error creating table: {e}, sql:\n{sql}", file=sys.stderr)
+        raise e
+
+
+# Создание таблицы
+def create_codeowners_table(session, table_path):
+    sql = f"""
+    --!syntax_v1
+    CREATE TABLE IF NOT EXISTS `{table_path}` (
+        path Utf8,
+        owners Utf8,
+        PRIMARY KEY (path)
+    );
+    """
+    create_table(session, sql)
+
+
+def create_tests_table(session, table_path):
+    # Создание таблицы, если ее еще нет
+    sql = f"""
+    --!syntax_v1
+    CREATE TABLE IF NOT EXISTS `{table_path}` (
+        build_type Utf8,
+        job_name Utf8,
+        job_id Uint64,
+        commit Utf8,
+        branch Utf8,
+        pull Utf8,
+        run_time Timestamp,
+        test_id Utf8,
+        suite_name Utf8,
+        raw_test_name Utf8,
+        file Utf8,
+        test_name Utf8,
+        fixture Utf8,
+        time Double,
+        status Utf8,
+        status_description Utf8,
+        owners Utf8,
+        log Utf8,
+        logsdir Utf8,
+        stderr Utf8,
+        stdout Utf8,
+        PRIMARY KEY(test_id)
+    );
+    """
+    create_table(session, sql)
+
+
+def upload_results(pool, sql):
+    with pool.checkout() as session:
+        try:
+            session.transaction().execute(sql, commit_tx=True)
+        except Exception as e:
+            print(f"Error executing query {e}, sql:\n{sql}", file=sys.stderr)
+            raise e
+
+
+def prepare_and_upload_tests(pool, path, results, batch_size):
+
+    total_records = len(results)
+    if total_records == 0:
+        print("Error:Object stored test results is empty, check test results artifacts")
+        return 1
+
+    with ThreadPoolExecutor(max_workers=10) as executor:
+
+        for start in range(0, total_records, batch_size):
+            futures = []
+            end = min(start + batch_size, total_records)
+            batch = results[start:end]
+
+            sql = f"""--!syntax_v1
+            UPSERT INTO `{path}` 
```

Comment:
I think it is better use bulk upsert here and do things in one query

Reply:
dont get your idea =(
im batching upserting tests results (for now batch size 50 tests in one request) + multithread (max 10 threads)
here it is https://github.com/ydb-platform/ydb/pull/6363/files/1ca678135650e68b42388d729c84cf3e578b41e6#diff-53571f121b3e2f183ba12b65c2d140f28249bb7d2cbcd1b812bc4cc5a654ed9eR258

Reply:
https://github.com/ydb-platform/ydb/blob/cc81c7eb0bdcd035ea66223a9d81bcc3c76f29ba/ydb/ci/build_bloat/ydb_upload.py#L182